asdf/_tests/test_reference.py

```diff
@@ -78,7 +78,9 @@ def do_asserts(ff):
 
         assert_array_equal(ff.tree["internal"], exttree["cool_stuff"]["a"])
 
-    with asdf.AsdfFile({}, uri=(tmp_path / "main.asdf").as_uri()) as ff:
+    with asdf.AsdfFile({}, uri=(tmp_path / "main.asdf").as_uri(), memmap=True) as ff:
+        # avoid passing tree to AsdfFile to avoid the deprecation warning, this can be updated
+        # when automatic find_references on AsdfFile.__init__ is removed
```

Comment:
My suggestion is to:
```suggestion
    with asdf.AsdfFile({}, uri=(tmp_path / "main.asdf").as_uri()) as ff:
```
and delete these lines:
https://github.com/asdf-format/asdf/blob/2ed29f0ad1a2452a90cff536fb3a0a0575ff3111/asdf/_tests/test_reference.py#L59-L61

The test on main is checking that an external reference to an array is memmapped. This is true when the default is `memmap=True` but is false with the default `memmap=False`. On main, when opening an `open(..., memmap=False)` the `memmap` argument is never passed to `open_external` (so it will always use the default).

I opened an issue for the pre-existing bug: https://github.com/asdf-format/asdf/issues/1867

For this PR I'd say we delete that portion of the test (as suggested above) since it doesn't check for anything specific to references (and failed to catch the above bug).